File: examples/migrate.py
from libpasta import *

class User:
    def auth_user(self):
        pw = read_password("Enter password")
        res = verify_password_update_hash(self.pw_hash, pw)
        if res.tag == HashUpdate.Updated:
            self.pw_hash = res.Updated
            return true
        elif res.tag == HashUpdate.Ok:
            return true
        else:
            return false

def migrate_users(users):
    for user in users:
        res = migrate_hash(user.pw_hash)
        if res.tag == HashUpdate.Updated:
            user.pw_hash = res.updated

File: examples/migrate.rb
require 'libpasta'

class User
    attr_accessor :pw_hash
    @pw_hash = ""


    def auth_user
        pw = Pasta::read_password("Enter password: ")
        res = Pasta::verify_password_update_hash(self.pw_hash, pw)
        case res.tag
        when Pasta::HashUpdate::Updated
            self.pw_hash = res.updated
            return true
        when  Pasta::HashUpdate::Ok
            return true
        else
            return false
        end
    end
end

def migrate_users(users)
    for user in users do
        res = Pasta::migrate_hash(user.pw_hash)
        if res.tag.eql? Pasta::HashUpdate::Updated
            user.pw_hash = res.updated
        end
    end
end

